contracts/foundation/service_provider_test.go

```diff
@@ -0,0 +1,37 @@
+package foundation
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+var testRegister = 0
+
+type test1ServiceProvider struct {
+}
```

Comment:
S: We can remove this struct.

Reply:
```go
var sp1 = &test1ServiceProvider{}
_, ok1 := interface{}(sp1).(ServiceProvider)
assert.False(t, ok1)
```

Isn't this kind of reverse testing necessary?

Reply:
Yes, it must be false, `test1ServiceProvider` is an empty struct.

Reply:
How about that?

```go
_, ok1 := interface{}(struct{}{}).(ServiceProvider)
assert.False(t, ok1)
```

Reply:
Done